unsteady/test_cases/trench_1d_der/options.py

```diff
@@ -0,0 +1,184 @@
+
+        self.eta_d = Function(self.P1DG).project(self.elev_init)
+
+        self.sediment_model = SedimentModel(
+            options=ModelOptions2d, suspendedload=self.suspended, convectivevel=self.convective_vel_flag,
+            bedload=self.bedload, angle_correction=self.angle_correction,
+            slope_eff=self.slope_eff, seccurrent=False, mesh2d=mesh, bathymetry_2d=bathymetry,
+            uv_init=self.uv_d, elev_init=self.eta_d, ks=self.ks, average_size=self.average_size,
+            cons_tracer=self.conservative, wetting_and_drying=self.wetting_and_drying
+        )
+
+    def set_quadratic_drag_coefficient(self, fs):
+        self.depth = Function(fs).interpolate(self.set_bathymetry(fs) + Constant(0.397))
+        if self.friction == 'nikuradse':
+            return interpolate(self.get_cfactor(self.depth), fs)
+
+    def get_cfactor(self, depth):
+        try:
+            assert hasattr(self, 'depth')
+        except AssertionError:
+            raise ValueError("Depth is undefined.")
+        ksp = Constant(3*self.average_size)
+        hc = conditional(depth > Constant(0.001), depth, Constant(0.001))
+        aux = max_value(11.036*hc/ksp, 1.001)
+        return 2*(0.4**2)/(ln(aux)**2)
+
+    def set_bathymetry(self, fs):
+        initial_depth = Constant(0.397)
+        depth_riv = Constant(initial_depth - 0.397)
+        depth_trench = Constant(depth_riv - 0.15)
+        depth_diff = depth_trench - depth_riv
+        x, y = SpatialCoordinate(fs.mesh())
+        trench = conditional(
+            le(x, 5), depth_riv, conditional(
+                le(x, 6.5), (1/1.5)*depth_diff*(x-6.5) + depth_trench, conditional(
+                    le(x, 9.5), depth_trench, conditional(
+                        le(x, 11), -(1/1.5)*depth_diff*(x-11) + depth_riv, depth_riv))))
+        return interpolate(-trench, fs)
+
+    #def set_viscosity(self, fs):
+    #    self.viscosity = Function(fs)
+    #    self.viscosity.assign(self.base_viscosity)
+    #    return self.viscosity
+
+    def set_boundary_conditions(self, prob, i):
+        inflow_tag = 1
+        outflow_tag = 2
+        boundary_conditions = {
+            'shallow_water': {
+                inflow_tag: {'flux': Constant(-0.22)},
+                outflow_tag: {'elev': Constant(0.397)},
+            },
+            'sediment': {
+                inflow_tag: {'value': self.sediment_model.equiltracer}
+            }
+        }
+        return boundary_conditions
+
+    def set_initial_condition(self, prob):
+        u, eta = prob.fwd_solutions[0].split()
+        u.project(self.uv_init)
+        eta.project(self.elev_init)
+
+    def set_sediment_source(self, fs):
+        if self.suspended:
+            return self.sediment_model.ero_term
+        else:
+            return None
+
+    def set_sediment_sink(self, fs):
+        if self.suspended:
+            return self.sediment_model.depo_term
+        else:
+            return None
```

Comment:
The `else` statement can be dropped as `None` is returned by default.

Reply:
removed